src/Microsoft.VisualStudio.ProjectSystem.Managed.VS/ProjectSystem/VS/DotNetCoreProjectCompatibilityDetector.cs

```diff
@@ -0,0 +1,317 @@
+﻿// Copyright (c) Microsoft.  All Rights Reserved.  Licensed under the Apache License, Version 2.0.  See License.txt in the project root for license information.
+
+using System;
+using System.Collections.Generic;
+using System.ComponentModel.Composition;
+using System.Runtime.InteropServices;
+using System.Runtime.Versioning;
+using System.Threading.Tasks;
+using Microsoft.VisualStudio.ProjectSystem.Properties;
+using Microsoft.VisualStudio.ProjectSystem.VS.Interop;
+using Microsoft.VisualStudio.ProjectSystem.VS.UI;
+using Microsoft.VisualStudio.Settings;
+using Microsoft.VisualStudio.Shell;
+using Microsoft.VisualStudio.Shell.Interop;
+using Microsoft.VisualStudio.Threading;
+using Task = System.Threading.Tasks.Task;
+
+namespace Microsoft.VisualStudio.ProjectSystem.VS
+{
+    /// <summary>
+    /// <see cref="IDotNetCoreProjectCompatibilityDetector"/>
+    /// </summary>
+    [Export(typeof(IDotNetCoreProjectCompatibilityDetector))]
+    internal sealed class DotNetCoreProjectCompatibilityDetector : IDotNetCoreProjectCompatibilityDetector, IVsSolutionEvents, IVsSolutionLoadEvents, IDisposable
+    {
+        // The version's below this are compatible, versions above it are unsupported (stronger warning message and no don't show again option), equal
+        // to are "partial" so tooling should generally work, new features may not have tooling.
+        static Version s_partialSupportedVersion = new Version(2, 1);
+
+        const string LearnMoreFwlink = "https://go.microsoft.com/fwlink/?linkid=867264";
+        private const string SuppressDotNewCoreWarningKey = @"ManagedProjectSystem\SuppressDotNewCoreWarning";
+
+        [Guid("9B164E40-C3A2-4363-9BC5-EB4039DEF653")]
+        private class SVsSettingsPersistenceManager { }
+
+        enum CompatibilityLevel
+        {
+            Supported = 0,
+            Partial = 1,
+            NotSupported =2
+        }
+        [ImportingConstructor]
+        public DotNetCoreProjectCompatibilityDetector([Import(typeof(SVsServiceProvider))] IServiceProvider serviceProvider, Lazy<IProjectServiceAccessor> projectAccessor, 
+                                                      Lazy<IDialogServices> dialogServices, Lazy<IProjectThreadingService> threadHandling)
+        {
+            _serviceProvider = serviceProvider;
+            ProjectServiceAccessor = projectAccessor;
+            DialogServices = dialogServices;
+            ThreadHandling = threadHandling;
+        }
+
+        public async Task InitializeAsync()
+        {
+            await ThreadHandling.Value.SwitchToUIThread();
+
+            // Do nothing, don't hook into events if not the release channel
+            if(IsReleaseChannel())
```

Comment:
I thought Andrew had mentioned that we didn't want to have different behavior between Previews and RTWs.

Reply:
No.  Here's an excerpt from the spec:
(note: this will only appear in 15.6 RTW since 15.7 will be available at the time and we want to encourage people to stay on the edge where they’ll have the best experience)

Reply:
Yeah, but I think this may have changed yesterday - best to follow up and be sure.

Reply:
I just synced up with Andrew - we think we should change it to just always show the warning, even if it's preview.